odoo/orm/decorators.py

```diff
@@ -263,69 +288,39 @@ def method(self, args):
 
     """
     if method.__name__ == 'create':
-        return model_create_single(method)
-    method._api = 'model'
+        return model_create_multi(method)  # type: ignore
+    method._api = 'model'  # type: ignore
     return method
 
 
-def readonly(method: T) -> T:
+def readonly(method: C) -> C:
     """ Decorate a record-style method where ``self.env.cr`` can be a
         readonly cursor when called trough a rpc call.
 
             @api.readonly
             def method(self, args):
                 ...
-
     """
-    method._readonly = True
-    return method
-
-_create_logger = logging.getLogger(__name__ + '.create')
-
-
-@decorator
-def _model_create_single(create: CreateLegacyCallee[S], self: S, arg: ValuesType | list[ValuesType]) -> S:
-    # 'create' expects a dict and returns a record
-    if isinstance(arg, Mapping):
-        return create(self, arg)
-    if len(arg) > 1:
-        _create_logger.debug("%s.create() called with %d dicts", self, len(arg))
-    return self.browse().concat(*(create(self, vals) for vals in arg))
-
-
-def model_create_single(method: CreateLegacyCallee[S]) -> CreateCaller[S]:
-    """ Decorate a method that takes a dictionary and creates a single record.
-        The method may be called with either a single dict or a list of dicts::
-
-            record = model.create(vals)
-            records = model.create([vals, ...])
-    """
-    warnings.warn(
-        f"Since 18.0, `create` must be batched, an override in {method.__module__} is in single mode",
-        DeprecationWarning,
-        stacklevel=3,
-    )
-    wrapper = typing.cast('CreateCaller[S]', _model_create_single(method))  # pylint: disable=no-value-for-parameter
-    wrapper._api = 'model_create'
-    return wrapper
+    return attrsetter('_readonly', True)(method)
 
 
 @decorator
-def _model_create_multi(create: CreateCallee[S], self: S, arg: ValuesType | list[ValuesType]) -> S:
+def _model_create_multi(create: Callable[[T, list[ValuesType]], T], self: T, arg: ValuesType | list[ValuesType]) -> T:
     # 'create' expects a list of dicts and returns a recordset
+    # make it accept a dict as a parameter
     if isinstance(arg, Mapping):
         return create(self, [arg])
     return create(self, arg)
 
 
-def model_create_multi(method: CreateCallee[S]) -> CreateCaller[S]:
+def model_create_multi(method: Callable[[T, list[ValuesType]], T]) -> Callable[[T, list[ValuesType] | ValuesType], T]:
     """ Decorate a method that takes a list of dictionaries and creates multiple
         records. The method may be called with either a single dict or a list of
         dicts::
 
             record = model.create(vals)
             records = model.create([vals, ...])
     """
-    wrapper = typing.cast('CreateCaller[S]', _model_create_multi(method))  # pylint: disable=no-value-for-parameter
+    wrapper = _model_create_multi(method)
```

Comment:
What about simplifying this?
```suggestion
    def create(self: T, vals_list: list[ValuesType] | ValuesType) -> T:
        if isinstance(vals_list, Mapping):
            vals_list = [vals_list]
        return method(self, vals_list)
        
    create.__doc__ = method.__doc__
```

Reply:
Updated.